### PR TITLE
State machine refactor - part2

### PR DESCRIPTION
**Pull request by iulianpascalau**
- implemented all steps
- added unit tests

### CODE REVIEW

File: relay/ethToElrond/steps/waitForSignaturesStepsUnHappyPaths_test.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/relay"
+	"github.com/ElrondNetwork/elrond-eth-bridge/relay/ethToElrond"
+	"github.com/ElrondNetwork/elrond-eth-bridge/relay/ethToElrond/steps/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFlowAsLeaderWaitSigsForTransferQuorumNotReachedWithStubChecking(t *testing.T) {
```
Comment (bogdan-rosianu) [inline relay/ethToElrond/steps/waitForSignaturesStepsUnHappyPaths_test.go:13]: please name test files with the same pattern of the productive code files

File: relay/ethToElrond/constants.go [hunk 2]
```diff
@@ -3,6 +3,22 @@ package ethToElrond
 const (
 	// GetPending is the step definition for the pending transactions check
 	GetPending = "get pending transactions"
+
 	// ProposeTransfer is the step definition for the propose transfer operation
 	ProposeTransfer = "propose transfer"
+
+	// WaitForSignaturesForProposeTransfer is the step definition for the signature gathering process for propose transfer
+	WaitForSignaturesForProposeTransfer = "wait for signatures for transfer propose"
```
Comment (bogdan-rosianu) [inline relay/ethToElrond/constants.go:11]: I think we should use Present Continuous Tense for these steps. "waiting signatures for transfer propose", "executing transfer" and so on

File: relay/ethToElrond/steps/happyCaseFlows_test.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	bem.WasSetStatusExecutedOnSourceCalled = trueHandler
+	bem.IsQuorumReachedForProposeTransferCalled = trueHandler
+	bem.IsQuorumReachedForProposeSetStatusCalled = trueHandler
+}
+
+func TestGetPendingEndlessLoop(t *testing.T) {
+	t.Parallel()
+
+	bem := mock.NewBridgeExecutorMock()
+	bem.HasPendingBatchCalled = func() bool {
```
Comment (dragos-rebegea) [inline relay/ethToElrond/steps/happyCaseFlows_test.go:31]: replace with falseHandler

File: relay/ethToElrond/steps/stepsFactory.go [hunk 4]
```diff
@@ -12,18 +13,44 @@ func CreateSteps(executor BridgeExecutor) (relay.MachineStates, error) {
...
+	return createMachineStates(executor)
 }
 
-func createMachineStates(executor BridgeExecutor) relay.MachineStates {
+func createMachineStates(executor BridgeExecutor) (relay.MachineStates, error) {
 	machineStates := make(relay.MachineStates)
 
-	machineStates[ethToElrond.GetPending] = &getPendingStep{
-		bridge: executor,
+	steps := []relay.Step{
```
Comment (dragos-rebegea) [inline relay/ethToElrond/steps/stepsFactory.go:22]: maybe declare this inside CreateSteps and pass it as an argument to createMachineStates

File: relay/ethToElrond/steps/010getPendingStep.go [hunk 5]
```diff
@@ -10,16 +10,22 @@ type getPendingStep struct {
 }
 
 // Execute will execute this step returning the next step to be executed
-func (gps *getPendingStep) Execute() relay.StepIdentifier {
-	gps.bridge.GetPendingBatch()
-	if gps.bridge.HasPendingBatch() {
+func (step *getPendingStep) Execute() relay.StepIdentifier {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply relay/ethToElrond/steps/010getPendingStep.go:13]: done

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**